src/main/webapp/script.js

```diff
@@ -4,6 +4,19 @@ let markers = [];
 const blueIconUrl = "http://maps.google.com/mapfiles/ms/icons/blue-dot.png";
 const orangeIconUrl = "http://maps.google.com/mapfiles/ms/icons/orange-dot.png";
 
+const configUrl = new URL('/client-config', window.location.origin)
+configUrl.searchParams.append("key", "clientapikey")
+
+fetch(configUrl).then((response => {
+  response.text().then(apikey => {
+    var script = document.createElement('script');
+    script.src = 'https://maps.googleapis.com/maps/api/js?key=' + apikey + '&libraries=places&callback=initMap';
+    script.defer = true;
+
+    document.head.appendChild(script);
+  })
+}))
+
```

Comment:
Will this slow-down the initial load-up speed of the website?

Reply:
Currently I'm seeing around 100ms delay to load the config on my desktop. Loading the frontend config takes about the same time as making 1 query - so its around 12% increase in delay.
Do you think its worth keeping it in to save the hassle of having to deal with api key in source code?

Reply:
Nice, but I think having the restrictions on the API key would be sufficient. I'm on the fence on this. @henryken care to share your thoughts on this?